WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
Oh:

> I wouldn't build a big string, but an array of clauses that you `.join(" AND ")` at the end. If AR wasn't so primitive, you could use the much cleaner Hash syntax, but this isn't MongoDB...

That sounded like you had a nicer way of doing the conditional chaining. But using a Hash, just replaces this:

```ruby
if params[:foo]
  query = query.where(foo: params[:foo])
end

query.to_a
```

with this:

```ruby
if params[:foo]
  hash.merge!(foo: params[:foo])
end

Foo.where(hash).to_a
```

Almost no difference.

> The class would know how to do `Date.safe_parse`

[`dry-types`](http://dry-rb.org/gems/dry-types/) could be used for coercion. It's still pre-1.0, though.